infra/src/main/java/com/redhat/service/smartevents/infra/exceptions/mappers/ExternalUserExceptionMapper.java

```diff
@@ -25,16 +28,26 @@ public class ExternalUserExceptionMapper implements ExceptionMapper<ExternalUser
     @Override
     public Response toResponse(ExternalUserException e) {
         LOGGER.debug("Failure", e);
-        Optional<BridgeError> error = bridgeErrorService.getError(e);
+        Optional<BridgeError> error = bridgeErrorService.getError(e.getClass());
         ResponseBuilder builder = Response.status(e.getStatusCode());
         if (error.isPresent()) {
             ErrorResponse errorResponse = ErrorResponse.from(error.get());
             errorResponse.setReason(e.getMessage());
-            builder.entity(errorResponse);
+            builder.entity(ErrorsResponse.toErrors(errorResponse));
         } else {
-            LOGGER.warn("Information for exception type {} cannot be found", e.getClass());
-            builder.entity(e.getMessage());
+            builder.entity(ErrorsResponse.toErrors(unmappedException(e)));
```

Comment:
@manstis Rather than mapping to an `UnclassifiedException`, should we just map to the generic `ExternalUserException` or perhaps the `BadRequestException` if there is no specific entry for the concrete class of the `Exception` being handled here?

So the logic becomes:

- Check to see if there is a specific override for the concrete `Exception` at runtime
- If yes, use that to return the error to the caller of the API
- If no, log a warning about an unmapped exception (including the exception class) and return the generic `ExternalUserException` Error.

If we did decide to go down the `ExternalUserException` route, I'd suggest we update the message for it in `exceptionInfo.json` which doesn't really make sense tbh.

To me the `UnclassifiedException` is indicative of a build-time problem for us. We need a process that determines that all `ExternalUserException` are mapped correctly into `exceptionInfo.json`, which might lend itself more to a Quarkus Plugin or _something_ like that.

WDYT?

Reply:
@robpblake All good suggestions. PR updated.